tests/nightly/gpu/test_bb2.py

```diff
@@ -8,56 +8,64 @@
-ZOO_BB2_3B = 'zoo:blenderbot2/blenderbot2_3B/model'
-SEARCH_SERVER = '<SERVER_API>'
-common_opt = {
-    'model': 'projects.blenderbot2.agents.blenderbot2:BlenderBot2RagAgent',
-    # rag args
-    'init_opt': 'arch/bart_large',
-    'generation_model': 'bart',
-    'retriever_debug_index': 'compressed',
-    'label_truncate': 128,
-    'text_truncate': 512,
-    'batchsize': 4,
-    'fp16': True,
-    'model_parallel': True,
-    # train args
-    'task': 'convai2,wizard_of_wikipedia',
-    'num_examples': 8,
-}
-
-
-def _test_bb2_rag(retrieval_method: KnowledgeAccessMethod, **kwargs):
-    opt = copy.deepcopy(common_opt)
-    opt['knowledge_access_method'] = retrieval_method.value
-    opt.update(dict(kwargs))
-    print(' '.join([f'--{k} {v}' for k, v in opt.items()]))
-    testing_utils.eval_model(opt, skip_test=True)
-    torch.cuda.empty_cache()
-
-
-def _test_bb2_fid(retrieval_method: KnowledgeAccessMethod, **kwargs):
-    opt = copy.deepcopy(common_opt)
-    opt['model'] = 'projects.blenderbot2.agents.blenderbot2:BlenderBot2FidAgent'
-    opt['knowledge_access_method'] = retrieval_method.value
-    opt.update(dict(kwargs))
-    testing_utils.eval_model(opt, skip_test=True)
-    torch.cuda.empty_cache()
+if TRANSFORMER_INSTALLED:
+    SEARCH_QUERY_MODEL = ZOO_MEMORY_DECODER
+    PERSONA_SUMMARY_MODEL = ZOO_QUERY_GENERATOR
+    ZOO_BB2 = 'zoo:blenderbot2/blenderbot2_400M/model'
+    ZOO_BB2_3B = 'zoo:blenderbot2/blenderbot2_3B/model'
+    SEARCH_SERVER = '<SERVER_API>'
+    common_opt = {
+        'model': 'projects.blenderbot2.agents.blenderbot2:BlenderBot2RagAgent',
+        # rag args
+        'init_opt': 'arch/bart_large',
+        'generation_model': 'bart',
+        'retriever_debug_index': 'compressed',
+        'label_truncate': 128,
+        'text_truncate': 512,
+        'batchsize': 4,
+        'fp16': True,
+        'model_parallel': True,
+        # train args
+        'task': 'convai2,wizard_of_wikipedia',
+        'num_examples': 8,
+    }
+
+    def _test_bb2_rag(retrieval_method: KnowledgeAccessMethod, **kwargs):
+        opt = copy.deepcopy(common_opt)
+        opt['knowledge_access_method'] = retrieval_method.value
+        opt.update(dict(kwargs))
+        print(' '.join([f'--{k} {v}' for k, v in opt.items()]))
+        testing_utils.eval_model(opt, skip_test=True)
+        torch.cuda.empty_cache()
+
+    def _test_bb2_fid(retrieval_method: KnowledgeAccessMethod, **kwargs):
+        opt = copy.deepcopy(common_opt)
+        opt['model'] = 'projects.blenderbot2.agents.blenderbot2:BlenderBot2FidAgent'
+        opt['knowledge_access_method'] = retrieval_method.value
+        opt.update(dict(kwargs))
+        testing_utils.eval_model(opt, skip_test=True)
+        torch.cuda.empty_cache()
 
 
 @testing_utils.skipUnlessGPU
 @unittest.skipIf(LOCAL, "Skipping Test because its slow and mem intensive")
+@unittest.skipUnless(TRANSFORMER_INSTALLED, "Needs transformer, not installed.")
```

Comment:
this is a bit funny because our GPU tests should always have transformers?

Reply:
yeah im positive this should not be necessary